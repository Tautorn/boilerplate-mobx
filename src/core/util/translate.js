export function setLanguage(language, reload) {
  localStorage.setItem('core.language', language)

  if (reload) {
    window.location.reload()
  }
}

export function getLanguage() {
  if (!localStorage.getItem('core.language')) {
    setLanguage(navigator.language, false)
  }

  return localStorage.getItem('core.language')
}

export function getPreferentialLanguage() {
  return window.navigator.languages[0] || window.navigator.language
}
