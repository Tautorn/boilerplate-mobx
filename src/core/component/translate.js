import { addLocaleData } from 'react-intl'
import pt from 'react-intl/locale-data/pt'
import en from 'react-intl/locale-data/en'
import * as Tools from 'core/util/translate'
import ptBR from 'core/i18n/pt-BR'
import enUS from 'core/i18n/en-US'

const Translate = {}

addLocaleData([...pt, ...en])
addLocaleData({ locale: 'pt-BR', parentLocale: 'pt' })
addLocaleData({ locale: 'en-US', parentLocale: 'en' })

const languageDefault = 'en-US'

const supported = { 'en-US': enUS, 'pt-BR': ptBR }

const preferential = Tools.getPreferentialLanguage()

const language = supported[preferential] ? preferential : languageDefault

Tools.setLanguage(language)

Translate.language = language
Translate.messages = supported[language]

export default Translate
