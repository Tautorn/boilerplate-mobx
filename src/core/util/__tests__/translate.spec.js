import * as translate from '../translate'

describe('Translate', () => {
  it('should get default language', () => {
    expect(translate.getLanguage()).toBe('en-US')
  })

  it('should set language', () => {
    translate.setLanguage('pt-BR')

    expect(translate.getLanguage()).toBe('pt-BR')
  })

  it('should reload after language change', () => {
    const reload = jest
      .spyOn(window.location, 'reload')
      .mockImplementation()

    translate.setLanguage('pt-BR', true)

    expect(reload).toHaveBeenCalled()

    reload.mockRestore()
  })

  it('should get preferencial language', () => {
    const language = translate.getPreferentialLanguage()
    const expected = ['en-US', 'pt-BR']

    expect(expected).toContain(language)
    
    expect(language).toMatchSnapshot()
  })

})
