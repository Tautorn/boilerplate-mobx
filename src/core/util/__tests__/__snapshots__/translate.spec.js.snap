// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Translate should get preferencial language 1`] = `"en-US"`;
